lib/src/helpers/lists.dart

```diff
@@ -3,29 +3,42 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:collection';
+import 'dart:js_interop';
 
-import '../../web.dart' show Touch, TouchList;
+/// Defines interop members on a JSObject that would be present on a js list object.
+/// The JSObject is assumed to be one of the js list type as right now
+/// we don't have a single interface that would represent all the js list types.
+extension on JSObject {
+  /// The **`item()`** method returns the [JSObject]
+  /// at the specified index in the list.
+  external JSObject item(int index);
 
-/// A wrapper to present a [TouchList] as a `List<Touch>`.
-class TouchListWrapper extends Object
-    with ListMixin<Touch>
-    implements List<Touch> {
-  final TouchList _original;
-  TouchListWrapper(this._original);
+  /// The **`length`** read-only property indicates the number of
+  /// items in a given list.
+  external int get length;
+}
+
+
+/// A wrapper to present a js list as a `List<T>`.
+class JSListWrapper<T extends JSObject> extends Object
+    with ListMixin<T>
+    implements List<T> {
+  final JSObject _original;
+  JSListWrapper(this._original);
 
   @override
   int get length => _original.length;
 
   @override
-  Touch operator [](int index) {
+  T operator [](int index) {
     if (index > length) {
       throw IndexError.withLength(index, length, indexable: this);
     }
-    return _original.item(index)!;
+    return _original.item(index) as T;
```

Comment:
I was initially actually trying to avoid adding another extension type but this def makes things much easier along with the changes you suggested above.

Reply:
Updated.